release-check.sh

```diff
@@ -0,0 +1,22 @@
+#!/usr/bin/env bash
+
+die () { echo "${1}"; exit 1; }
+
+# Check that the current HEAD is on origin/master.
+HEAD=`git rev-parse --verify HEAD`
+MASTERURL="https://api.github.com/repos/facebook/reason/git/refs/heads/master"
+HEADMSG="Current HEAD is not on upstream master. This is a requirement before releasing.
+If you are sure it is, try switching branches to master and pulling changes."
+echo "Checking HEAD against upstream master..."
+curl --silent $MASTERURL | grep "sha" | grep $HEAD || die "$HEADMSG"
```

Comment:
Do you think we should use this api instead? :https://developer.github.com/v3/repos/commits/#get-a-single-commit 

Looks like this API tells you if one commit is currently on master or not.